lib/http/response/body.rb

```diff
@@ -14,20 +14,30 @@ def initialize(client, encoding = Encoding::BINARY)
         @client    = client
         @streaming = nil
         @contents  = nil
-        @encoding  = encoding
+
+        # see issue 312
+        begin
+          @encoding = Encoding.find encoding
+        rescue ArgumentError
+          @encoding = Encoding::BINARY
+        end
       end
 
       # (see HTTP::Client#readpartial)
       def readpartial(*args)
         stream!
-        @client.readpartial(*args)
+        force_encoding @client.readpartial(*args)
```

Comment:
The downside of `String#force_encoding` vs `String#encoding` is it will allow invalid encodings through. Should we do that in the event the upstream responder produces invalid encoded output? It's leaving a sort of lazy string encoding time bomb... I'm curious if there's precedent for how this sort of thing is handled elsewhere